xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -5059,3 +5088,12 @@ int CVideoPlayer::GetProgramsCount()
   return m_programs.size();
 }
 
+void CVideoPlayer::SetUpdateStreamDetails(const CFileItem& item, bool update)
+{
+  if (IsRunning())
```

Comment:
no need to check this. if clients call this if not running, nothing will happen. that is the contract.

Reply:
Yes, nothing will happen, and m_UpdateStreamDetails will not be updated. If we call CVideoPlayer::OpenFile when VP thread is not running then on OnPlayBackStarted cb and on subsequent call to SetUpdateStreamDetails VP thread can be not running too. I have to check - if VP thread is running then I update m_UpdateStreamDetails with message, else update directly (we've got playback cb, then playback is starting and current file item is set, so we are allowed to update this). The same as in OpenFile. I've already tested it.
```
bool CVideoPlayer::OpenFile(const CFileItem& file, const CPlayerOptions &options)
{
  if (IsRunning())...
  // we're not running
  m_item = file;
  m_callback.OnPlayBackStarted(m_item);
  // this cb is calling SetUpdateStreamDetails that can be executed before starting VP thread
  //   m_messenger.Put(...) do nothing and m_UpdateStreamDetails is not updated
}
```

Reply:
VP should not fire OnPlabackStarted when not running. In OpenFIle, move the cb down after Create() and m_messenger.Init() from OnStartup to OpenFIle after Create()